### PR TITLE
[FSSDK-9028]: Adds native iOS implementation for ODP

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Adds native iOS implementation for ODP
- Updates iOS sdk version to `4.0.0-beta`

## Test plan
- All previous unit tests should pass.

## Issues
- [FSSDK-9028](https://jira.sso.episerver.net/browse/FSSDK-9028)

### CODE REVIEW

File: ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift [hunk 1]
```diff
@@ -67,6 +67,14 @@ public class SwiftOptimizelyFlutterSdkPlugin: NSObject, FlutterPlugin {
         case API.removeForcedDecision: removeForcedDecision(call, result: result)
         case API.removeAllForcedDecisions: removeAllForcedDecisions(call, result: result)
         case API.close: close(call, result: result)
+            
+            // ODP
```
Comment (Mat001) [inline ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift:71]: fix comment indent

File: ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift [hunk 2]
```diff
@@ -359,6 +394,108 @@ public class SwiftOptimizelyFlutterSdkPlugin: NSObject, FlutterPlugin {
         result(createResponse(success: true))
     }
     
+    /// Returns an array of segment names that the user is qualified for.
```
Comment (Mat001) [inline ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift:397]: can probably say just "Returns an array of segments that the user..."

File: ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift [hunk 3]
```diff
@@ -359,6 +394,108 @@ public class SwiftOptimizelyFlutterSdkPlugin: NSObject, FlutterPlugin {
...
+        if let _type = parameters[RequestParameterKey.type] as? String {
+            type = _type
+        }
+        if let _identifiers = parameters[RequestParameterKey.identifiers] as? Dictionary<String, String> {
+            identifiers = _identifiers
+        }
+        if let _data = Utils.getTypedMap(arguments: parameters[RequestParameterKey.data] as? Any) {
+            data = _data
+        }
+
```
Comment (Mat001) [inline ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift:471]: Do all these type definitions guard against invalid cases? What identifiers are empty?

**Review by Mat001 — APPROVED** [review]
Looks ok.

File: ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift (hunk 3 above)
Comment (yasirfolio3) [reply ios/Classes/SwiftOptimizelyFlutterSdkPlugin.swift:471]: yes, this makes sure only valid data is received from flutter. Data is empty by default.

**Review by Mat001 — APPROVED** [review]
lgtm

**Pull request merged**